src/main/java/uk/gov/hmcts/reform/security/keyvault/KeyVaultService.java

```diff
@@ -85,87 +114,125 @@ protected KeyVaultClient getClient() {
      * @should call delegate
      */
     KeyBundle getKeyByAlias(String alias) {
-        return keyByAliasCache.getUnchecked(alias);
+        alias = replaceDotsWithDashes(alias);
+        return getFromCacheOrNull(keyByAliasCache::getUnchecked, alias);
     }
 
     /**
      * @should call delegate
      */
-    KeyBundle getKeyByIdentifier(String keyIdentifier) {
-        return keyByIdentifierCache.getUnchecked(keyIdentifier);
+    SecretBundle getSecretByAlias(String alias) {
+        alias = replaceDotsWithDashes(alias);
+        return getFromCacheOrNull(secretByAliasCache::getUnchecked, alias);
     }
 
     /**
-     * @should call delegate
+     * @should call delegate if key is SecretKey
+     * @should throw exception if key is unsupported
      */
-    CertificateBundle getCertificateByAlias(String alias) {
-        return certificateByAliasCache.getUnchecked(alias);
+    public SecretBundle setKeyByAlias(String alias, Key key) {
+        alias = replaceDotsWithDashes(alias);
+        if (key instanceof SecretKey) {
+            JsonWebKey jsonWebKey = JsonWebKey.fromAes((SecretKey) key);
+            SetSecretRequest secretRequest = new SetSecretRequest
+                .Builder(baseUrl, alias, new String(jsonWebKey.k()))
+                .build();
+            return this.vaultClient.setSecret(secretRequest);
+        }
+        throw new UnsupportedOperationException("Only SecretKey Operations have been implemented");
     }
 
     /**
-     * @should call delegate
+     * @should call delegate and return parsed list
      */
-    KeyOperationResult sign(String keyIdentifier, JsonWebKeySignatureAlgorithm algorithm, byte[] digest) {
-        return vaultClient.sign(keyIdentifier, algorithm, digest);
+    public List<String> engineAliases() {
```

Comment:
Thinking about it, it would be nice to add certificate aliases to the returned list as well - this might help us with a specific issue we have when using the JCA provider in ForgeRock AM (name of the signing certificate is hardcoded to "test").

Reply:
added